app/include/zmk/events/keycode-state-changed.h

```diff
@@ -18,14 +18,16 @@ struct keycode_state_changed {
     u32_t keycode;
     u8_t implicit_modifiers;
     bool state;
+    s64_t timestamp;
```

Comment:
So, this refactor is mixed in with the feature itself. I'm not opposed to that being in this PR, but can we at least break that into a sequence of two commits, the refactor, and then the feature, so we can review this a bit more easily?